src/core/types/base/mempool.ts

```diff
@@ -98,17 +107,50 @@ export function decodeMempool(mempool: Mempool, ignoreAddresses: Record<string,
  * @param ignoreAddresses: Array containing wallet addresses that should be ignored and filtered.
  * @returns Boolean stating if `msg` is to be processed or not.
  */
-function isAllowedMempoolMsg(msg: MsgExecuteContract, ignoreAddresses: Record<string, boolean>): boolean {
+function isAllowedMempoolMsg(
+	msg: MsgExecuteContract,
+	ignoreAddresses: IgnoredAddresses,
+	timeoutDur: number,
+	iteration: number,
+): boolean {
 	//if the sender of the message is in our ignore list: skip this message
 	if (ignoreAddresses[msg.sender]) {
-		return false;
+		return !keepIgnored(msg.sender, undefined, ignoreAddresses, timeoutDur, iteration);
```

Comment:
its vague to return !function(), maybe rework this a bit so you can return the function instead of its opposite